script/tool/lib/src/update_dependency_command.dart

```diff
@@ -181,6 +226,58 @@ ${response.httpResponse.body}
     return PackageResult.success();
   }
 
+  /// Handles all of the updates for [package] when the target dependency is
+  /// an Android dependency.
+  Future<PackageResult> _runForAndroidDependency(
+      RepositoryPackage package) async {
+    if (_targetAndroidDependency == 'gradle') {
+      final Iterable<RepositoryPackage> packageExamples = package.getExamples();
+      for (final RepositoryPackage example in packageExamples) {
+        if (!example.directory.childDirectory('android').existsSync()) {
+          return PackageResult.skip('Example app does not run on Android.');
+        }
+
+        Directory gradleWrapperPropertiesDirectory =
+            example.directory.childDirectory('android');
+        if (gradleWrapperPropertiesDirectory
+            .childDirectory('app')
+            .childDirectory('gradle')
+            .existsSync()) {
+          gradleWrapperPropertiesDirectory =
+              gradleWrapperPropertiesDirectory.childDirectory('app');
+        }
+        final File gradleWrapperPropertiesFile =
+            gradleWrapperPropertiesDirectory
+                .childDirectory('gradle')
+                .childDirectory('wrapper')
+                .childFile('gradle-wrapper.properties');
+
+        final String gradleWrapperPropertiesContents =
+            gradleWrapperPropertiesFile.readAsStringSync();
+        final RegExp validGradleDistributionUrl =
+            RegExp(r'^\s*distributionUrl\s*=\s*.*\.zip', multiLine: true);
+        if (!validGradleDistributionUrl
+            .hasMatch(gradleWrapperPropertiesContents)) {
+          return PackageResult.fail(<String>[
+            'Failed to update Gradle version in the example app for ${package.displayName}'
+          ]);
+        }
+
+        final String newGradleWrapperPropertiesContents =
+            gradleWrapperPropertiesContents.replaceFirst(
+                validGradleDistributionUrl,
+                // ignore: unnecessary_string_escapes
+                'distributionUrl=https\://services.gradle.org/distributions/gradle-$_targetVersion-all.zip');
+        gradleWrapperPropertiesFile
+            .writeAsStringSync(newGradleWrapperPropertiesContents);
```

Comment:
We try to be relatively verbose in this tooling so it's easy for people running it to see what the commands are actually doing; we should print something for each entry we change (along the lines of line 201 above).